### PR TITLE
circonus_annotation: fix broken import, update requirements

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
Fix [broken import](https://github.com/ansible/community/wiki/Testing%3A-progress-tracker#fixing-broken-imports) in `circonus_annotation` module

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
monitoring/circonus_annotation.py

##### ANSIBLE VERSION
```
ansible-playbook 2.4.0 (devel 7bc85f9b44) last updated 2017/07/14 11:19:12 (GMT +200)
```

### CODE REVIEW

File: lib/ansible/modules/monitoring/circonus_annotation.py [hunk 1]
```diff
@@ -91,7 +89,11 @@
 import json
 import time
 
-import requests
+try:
+    import requests
+    HAS_REQUESTS = True
```
Comment (bcoca) [inline lib/ansible/modules/monitoring/circonus_annotation.py:147]: requests is a 'fun' library, the API has changed over time, so you should always check for a specific version (I know this is an issue with the original module, but that should have been caught before merging it.)

Comment (pilou-) [reply lib/ansible/modules/monitoring/circonus_annotation.py:147]: I made some tests:

* using Python 3: requests `2.0.0` is required, there is an error with both `1.0.0` and `1.2.3`:
* using Python 2, the module works fine using `1.0.0`, `2.0.0`, `2.18.13`, there is an error with `0.14.2` (at this time, `response.json` [wasn't](https://media.readthedocs.org/pdf/requests/v0.14.2/requests.pdf) a method).

Error using Python 3 and requests < 2.0.0:

```
lib/ansible/modules/monitoring/circonus_annotation.py:155: in main
    resp = post_annotation(annotation, module.params['api_key'])
lib/ansible/modules/monitoring/circonus_annotation.py:106: in post_annotation
    headers=build_headers(api_key), data=json.dumps(annotation))
../../../.venv/ansible3_tests/lib/python3.5/site-packages/requests/api.py:88: in post
    return request('post', url, data=data, **kwargs)
../../../.venv/ansible3_tests/lib/python3.5/site-packages/requests/api.py:44: in request
    return session.request(method=method, url=url, **kwargs)
../../../.venv/ansible3_tests/lib/python3.5/site-packages/requests/sessions.py:335: in request
    resp = self.send(prep, **send_kwargs)
../../../.venv/ansible3_tests/lib/python3.5/site-packages/requests/sessions.py:438: in send
    r = adapter.send(request, **kwargs)
../../../.venv/ansible3_tests/lib/python3.5/site-packages/requests/adapters.py:292: in send
    timeout=timeout
../../../.venv/ansible3_tests/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py:423: in urlopen
    conn = self._get_conn(timeout=pool_timeout)
../../../.venv/ansible3_tests/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py:238: in _get_conn
    return conn or self._new_conn()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    def _new_conn(self):
[...]
        connection = VerifiedHTTPSConnection(host=self.host,
                                             port=self.port,
>                                            strict=self.strict)
E       TypeError: __init__() got an unexpected keyword argument 'strict'
```

**Comment by pilou-** [comment]
Pull request updated:
- requirements updated
- boilerplate added and `get_exception` removed
- unit tests added

Related: #26832 (circonus_annotation documentation updates)

File: lib/ansible/modules/monitoring/circonus_annotation.py [hunk 2]
```diff
@@ -151,12 +143,15 @@ def main():
             api_key=dict(required=True, no_log=True)
             )
         )
+
+    if not HAS_REQUESTS:
+        module.fail_json(msg='requests is required for this module')
```
Comment (abadger) [inline lib/ansible/modules/monitoring/circonus_annotation.py]: I think @bcoca was suggesting that there should be a version check in the code.  Something like:
``` python
from distutils.version import LooseVersion
from ansible.module_utils.six import PY3

[...]

def check_requests_dep(module):
    if not HAS_REQUESTS:
        module.fail_json(msg='requests (version 1.0+ for Python2 or 2.0+ for Python3) is required for this module')

    if PY3:
        if LooseVersion(requests.__version__) < LooseVersion('2.0'):
            module.fail_json(msg='On Python3, this module requires requests-2.0 or higher')
    else:
        if LooseVersion(requests.__version__) < LooseVersion('1.0'):
            module.fail_json(msg='On Python2, this module requires requests-1.0 or higher')

[....]

# in def main():
check_requests_dep(module)
```

Comment (pilou-) [reply lib/ansible/modules/monitoring/circonus_annotation.py]: Done

**Comment by pilou-** [comment]
Pull request updated:
- rebased
- ~~boilerplate added and `get_exception` removed~~
- check `requests.__version__`

Related: #26832 (circonus_annotation documentation updates)

**Comment by abadger** [comment]
the new unittests are failing.  Looks like the exception raised isn't what the test expects.

**Comment by pilou-** [comment]
@abadger indeed, `reason` was an exception and become a string (https://github.com/ansible/ansible/commit/7cd81b802de061143e596823d86b9726b60a7ccb#diff-cf028138e58373672b67a512c3e71f8dL159). It seems `reason` should have been a string from the beginning, thanks for catching that !

Pull request updated:
- rebased
- test updated, `reason` is now a string

I am waiting CI results before adding `ready-for-review` :)

**Comment by abadger** [comment]
passed in shippable!  Thanks @pilou- , merged to devel for 2.4.0 release.

**Pull request merged**